### PR TITLE
DTO and Mapper for menu items

### PR DESCRIPTION
**Pull request by vjwua**
The code itself seems to be working, but the request is bad.

### CODE REVIEW

File: src/main/java/com/pnudev/springonlinedelivery/controllers/MenuItemController.java [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
 
-import com.pnudev.springonlinedelivery.models.MenuItem;
+import com.pnudev.springonlinedelivery.dto.MenuItemDto;
+import com.pnudev.springonlinedelivery.repos.MenuItemRepository;
 import com.pnudev.springonlinedelivery.service.MenuItemService;
 import lombok.AllArgsConstructor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (justme1135) [inline src/main/java/com/pnudev/springonlinedelivery/controllers/MenuItemController.java]: rewrite to single imports

File: src/main/java/com/pnudev/springonlinedelivery/service/MenuItemServiceImpl.java [hunk 2]
```diff
@@ -1,17 +1,69 @@
...
+    public void putMenuItem(MenuItem menuItem) {
+        // TODO TBW
+    }
+
+    @Override
+    public void cancelMenuItem(MenuItem menuItem) {
+        // TODO TBW
+    }
+
+    public MenuItem dtoToEntity(MenuItemDto menuItemDto) {
```
Comment (justme1135) [inline src/main/java/com/pnudev/springonlinedelivery/service/MenuItemServiceImpl.java]: You do not need to write mappers explicitly, `mapstruct` does this job for you

File: src/main/java/com/pnudev/springonlinedelivery/service/MenuItemServiceImpl.java [hunk 3]
```diff
@@ -1,17 +1,69 @@
...
+        MenuItem menuItem = new MenuItem();
+        menuItem.setName(menuItemDto.getName());
+        menuItem.setSummary(menuItemDto.getSummary());
+        menuItem.setSize(menuItemDto.getSize());
+        menuItem.setPrice(menuItemDto.getPrice());
+        menuItem.setOrder(menuItemDto.getOrder());
+        return menuItem;
+    }
+
+    public MenuItemDto entityToDto(MenuItem menuItem) {
```
Comment (justme1135) [inline src/main/java/com/pnudev/springonlinedelivery/service/MenuItemServiceImpl.java]: same here

**Review by justme1135 — COMMENTED** [review]
Overall I cannot see some specific problem with that. Please provide the exact endpoint, JSON request and error log from where you are getting BAD_REQUEST

**Comment by vjwua** [comment]
I got the error, it was associated with scanBasePackages.

**Review by justme1135 — APPROVED** [review]

**Pull request merged**